api/src/main/java/org/pmiops/workbench/api/OfflineWorkspaceController.java

```diff
@@ -0,0 +1,71 @@
+package org.pmiops.workbench.api;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.inject.Provider;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeComparator;
+import org.pmiops.workbench.config.WorkbenchConfig;
+import org.pmiops.workbench.db.dao.WorkspaceDao;
+import org.pmiops.workbench.db.model.DbWorkspace;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class OfflineWorkspaceController implements OfflineWorkspaceApiDelegate {
+
+  private final WorkspaceDao workspaceDao;
+  private final Provider<WorkbenchConfig> configProvider;
+
+  @Autowired
+  OfflineWorkspaceController(Provider<WorkbenchConfig> configProvider, WorkspaceDao workspaceDao) {
+    this.configProvider = configProvider;
+    this.workspaceDao = workspaceDao;
+  }
+
+  @Override
+  public ResponseEntity<Void> updateResearchPurposeReview() {
+    if (configProvider.get().featureFlags.enableResearchPurposePrompt) {
+      List<DbWorkspace> workspaceList = workspaceDao.findAllByReviewResearchPurpose((short) 1);
+
+      // 1. Filter out workspace where creation date is 365 days or 15 days
+      // 2. set Research Purpose Review to FALSE
+      // 3. Update workspace
+      workspaceList =
+          workspaceList.stream()
+              .filter(
+                  workspace -> {
+                    DateTime workspaceCreationDate = new DateTime(workspace.getCreationTime());
+                    DateTime workspaceModifiedDate = new DateTime(workspace.getLastModifiedTime());
+                    return checkReviewDate(workspaceCreationDate, workspaceModifiedDate);
+                  })
+              .map(
+                  filteredWorkspace -> {
+                    filteredWorkspace.setResearchPurposeReviewed(false);
+                    return filteredWorkspace;
+                  })
+              .collect(Collectors.toList());
+      workspaceDao.save(workspaceList);
+    }
+    return ResponseEntity.noContent().build();
+  }
+
+  private boolean checkReviewDate(DateTime workspaceCreationDate, DateTime workspaceModifiedDate) {
+    DateTime creationDatePlus15 = workspaceCreationDate.plusDays(15);
+    DateTime creationDatePlus1Year = workspaceCreationDate.plusYears(1);
+
+    // True if
+    // 1. Current date is creation Date + 1 year or
+    // 2. Current date is 15 days after creation date or
+    // 3. Current Date is AFTER an year of creation date (in case cron job did not run/ missed
+    // picking the workspace) and the Workspace has
+    // not been modified after 1 year of creation date.
+    int compareCreationDatePlus1Year =
```

Comment:
We need to update the story (will do it right now) the logic is as follows now and please let me know if it matches the understanding from code: it just checks if its more than 15 days/365 days. We are not implementing the update logic.

So this cron job will run and start marking workspaces that matches the days criteria

Reply:
I didn't realize the requirements had been updated, thank you for letting me know.

The logic still won't work. You're comparing the creation date plus one year / plus fifteen days against nothing. You need to compare the creation date plus one year / plus fifteen days against the current date.

Reply:
null means now
**
     * Compare two objects against only the range of date time fields as
     * specified in the constructor.
     * 
     * @param lhsObj  the first object,
     *      logically on the left of a &lt; comparison, null means now
     * @param rhsObj  the second object,
     *      logically on the right of a &lt; comparison, null means now
     * @return zero if order does not matter,
     *      negative value if lhsObj &lt; rhsObj, positive value otherwise.

Reply:
oh my god that is the worst way for a library to default a thing that I have ever seen.

Alright, this is fine, but can you leave a comment saying that null means now with a hyperlink to the documentation for this method call, please?